src/EnergyPlus/HVACInterfaceManager.cc

```diff
@@ -691,13 +691,24 @@ namespace HVACInterfaceManager {
 
         } else { // tank has mass
             if (MassFlowRate > 0.0) {
-                TankFinalTemp = (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
-                                    std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds) +
-                                (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp);
-                TankAverageTemp = ((ThisTankMass * Cp) / (MassFlowRate * Cp) *
-                                       (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
-                                       (1.0 - std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds)) / TimeStepSeconds +
-                                   (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp));
+                Real64 ExponentTerm = (MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds;
+                if (ExponentTerm >= 700.0) {
+                    TankFinalTemp = (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp);
+
+                    TankAverageTemp =
+                        ((ThisTankMass * Cp) / (MassFlowRate * Cp) *
+                             (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) / TimeStepSeconds +
+                         (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp));
+                } else {
+                    TankFinalTemp = (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
+                                        std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds) +
+                                    (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp);
+
+                    TankAverageTemp = ((ThisTankMass * Cp) / (MassFlowRate * Cp) *
+                                           (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
+                                           (1.0 - std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds)) / TimeStepSeconds +
+                                       (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp));
+                }
```

Comment:
This is really messy.  I wonder if there is anything you could evaluate to clean this up.  Maybe make a couple intermediate variables?

`Real64 const mdotCp = MassFlowRate * Cp;`

That alone may reduce the visual complexity of this code.  If there is another obvious choice, that would be great, but I understand it's a balance.

Reply:
Implemented suggestion by defining local variables.